feathr_project/feathr/client.py

```diff
@@ -399,12 +401,16 @@ def _get_offline_features_with_config(self, feature_join_conf_path='feature_join
                                                        feature_config=os.path.join(self.local_workspace_dir, 'feature_conf/'),
                                                        job_output_path=feathr_feature['outputPath'],
                                                        )
+        job_tags = {OUTPUT_PATH_TAG:feature_join_job_params.job_output_path}
+        # set output format in job tags if it's set by user, so that it can be used to parse the job result in the helper function
+        if execution_configuratons is not None and OUTPUT_FORMAT in execution_configuratons:
+            job_tags[OUTPUT_FORMAT]= execution_configuratons[OUTPUT_FORMAT]
 
         # submit the jars
         return self.feathr_spark_laucher.submit_feathr_job(
             job_name=self.project_name + '_feathr_feature_join_job',
             main_jar_path=self._FEATHR_JOB_JAR_PATH,
-            job_tags={OUTPUT_PATH_TAG:feature_join_job_params.job_output_path},
+            job_tags=job_tags,
             main_class_name='com.linkedin.feathr.offline.job.FeatureJoinJob',
             arguments=[
```

Comment:
why are we using tgs and some other times using arguments to control the behaviour? We should standardize.

Reply:
ya let me update. we should standardize

Reply:
This is a good point and I should also document this in the code. To make it a bit clear:
- Job tags are for job metadata and it's not passed to the actual spark job (i.e. not visible to spark job), more like a platform related thing that Feathr want to add (currently job tags only have job output URL and job output format, ). They are carried over with the job and is visible to every Feathr client. Think this more like some customized metadata for the job which would be weird to be put in the spark job itself.
- Job arguments (or sometimes called job parameters)are the arguments which are command line arguments passed into the actual spark job. This is usually highly related with the spark job. In Feathr it's like the input to the scala spark CLI. They are usually not spark specific (for example if we want to specify the location of the feature files, or want to 
- Job configuration are like "configurations" for the spark job and are usually spark specific. For example, we want to control the no. of write parts for spark

Job configurations and job arguments (or sometimes called job parameters) have quite some overlaps (i.e. you can achieve the same goal by either using the job arguments/parameters vs. job configurations). But the job tags should just be used for metadata purpose.